generate-workflow.py

```diff
@@ -323,92 +324,173 @@ def __str__(self):
+                            ''') + convert_flags))
+
+        if generate_stats:
+            perf_dir_name = "3c_performance_stats/"
+            steps.append(
+                Step(
+                    'Copy 3c stats of ' + component_friendly_name,
+                    textwrap.dedent(f'''\
+                                cd {component_dir}
+                                mkdir {perf_dir_name}
+                                cp *.json {perf_dir_name}
+                                ''')))
+            perf_artifact_name = component_friendly_name + job_suffix + ('_alltypes' if alltypes else '_noalltypes')
+            perf_dir = os.path.join(component_dir, perf_dir_name)
+            steps.append(
+                ActionStep(
+                    'Upload 3c stats of ' + component_friendly_name,
+                    'actions/upload-artifact@v2',
+                    {'name': perf_artifact_name, 'path': perf_dir, 'retention-days': 5}))
+
+        steps.append(
+            Step(
+                'Build converted ' + component_friendly_name +
+                at_ignore_step,
+                # convert_project.py sets -output-dir=out.checked as
+                # standard.
+                textwrap.dedent(f'''\
+                            cd {component_dir}
+                            cp -r out.checked/* .
+                            rm -r out.checked
+                            ''') +
+                #
+                (f'cd {component.build_dir}\n'
+                 if component.build_dir is not None else '') +
+                f'{build_converted_cmd}{at_ignore_code}\n'))
+
+    # We want blank lines between steps but not after the last step of
+    # the last benchmark.
+    out.write('\n'.join(str(s) for s in steps) + '\n')
+
+
+regular_run_configurations = {
+    False: [],
+    True: []
+}
+
+exhaustive_run_configurations = {
+    False: ['-only-g-sol', '-only-l-sol'],
+    True: []
+}
+
+'''
+Format:
+  {
+    workflow_file_path : (<workflow_name>, <run_config>, [True|False] <- flag to enable stats generation)
+  }
+'''
+workflow_file_configs = {
+    '.github/workflows/main.yml': ("3C benchmark tests",
+                                   regular_run_configurations, False),
+    '.github/workflows/exhaustive.yml': ("Exhaustive testing and Performance Stats",
+                                         exhaustive_run_configurations, True)
+}
+
+for workflow_file in workflow_file_configs:
+    workflow_name = workflow_file_configs[workflow_file][0]
+    run_configurations = workflow_file_configs[workflow_file][1]
+    generate_stats = workflow_file_configs[workflow_file][2]
+    with open(workflow_file, 'w') as out:
+        out.write(HEADER.replace('{workflow.name}', workflow_name))
+        for binfo in benchmarks:
+            for alltypes in run_configurations:
+                create_benchmark(out, binfo, alltypes, generate_stats)
+                extra_args = run_configurations[alltypes]
+                if len(extra_args) > 0:
```

Comment:
Remove this conditional: if `len(extra_args) == 0`, the for loop does nothing anyway.